src/SonarScanner.MSBuild.Common/EnvironmentBasedPlatformHelper.cs

```diff
@@ -19,19 +19,54 @@
  */
 
 using System;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
 
 namespace SonarScanner.MSBuild.Common;
 
 public class EnvironmentBasedPlatformHelper : IPlatformHelper
 {
+    private readonly Lazy<IPlatformHelper.OS> operatingSystem = new(CurrentOperatingSystem);
+
+    public IPlatformHelper.OS OperatingSystem => operatingSystem.Value;
     public static IPlatformHelper Instance { get; } = new EnvironmentBasedPlatformHelper();
 
     private EnvironmentBasedPlatformHelper()
     {
     }
 
     public string GetFolderPath(Environment.SpecialFolder folder, Environment.SpecialFolderOption option) => Environment.GetFolderPath(folder, option);
-    public bool DirectoryExists(string path) => System.IO.Directory.Exists(path);
-    public bool IsWindows() => Environment.OSVersion.Platform == PlatformID.Win32NT;
-    public bool IsMacOSX() => Environment.OSVersion.Platform == PlatformID.MacOSX;
+    public bool DirectoryExists(string path) => Directory.Exists(path);
+
+    [ExcludeFromCodeCoverage]
+    private static IPlatformHelper.OS CurrentOperatingSystem()
+    {
+        if (Environment.OSVersion.Platform == PlatformID.Win32NT)
+        {
+            return IPlatformHelper.OS.Windows;
+        }
+        else if (IsMacOs())
+        {
+            return IPlatformHelper.OS.MacOSX;
+        }
+        // Note: the Check for Mac Os must preceed the check for Unix, because Environment.OSVersion.Platform returns PlatformID.Unix on Mac Os
+        else if (Environment.OSVersion.Platform == PlatformID.Unix)
+        {
+            return IPlatformHelper.OS.Unix;
+        }
+        else
+        {
+            return IPlatformHelper.OS.Unknown;
+        }
+    }
+
+    // RuntimeInformation.IsOSPlatform is not suported in .NET Framework 4.6.2, it's only available from 4.7.1
+    // SystemVersion.plist exists on Mac Os (and iOS) for a very long time (at least from 2002), so it's safe to check it, even though it's not a pretty solution.
```

Comment:
Personal opinion: `for a very long time (at least from 2002)` -> `at least since 2002`. It's shorter and conveys the same meaning.